docs/language.rst

```diff
@@ -177,6 +177,122 @@ Lists of primitive types are also primitive types: ``string[]``, ``number[]``, `
     # incorrect, can't assign to dict after construction
     # a["otherkey"] = "othervalue"
 
+Strings
++++++++
+
+
+There are four kinds of strings in the Inmanta language:
+
+- regular strings
+
+.. code-block:: inmanta
```

Comment:
This code block introduces a lexing error in the documentation build:

```
language.rst:188: WARNING: Lexing literal_block 'regular_string_1 = "This is...\\n...a basic string."\n\n# Expected output e.g. when displayed:\n# This is...\n# ...a basic string.\n\n\nregular_string_2 = \'This one too.\'\n\n# Expected output e.g. when displayed:\n# This one too.' as "inmanta" resulted in an error at token: "'". Retrying in relaxed mode.
```

Reply:
(Maybe this is related to the red boxes around the quotes)

Reply:
That's weird, this snippet compiles just fine. I noticed another case of a lexing error resulting in red boxes. But it looks like the red boxes only appear in my local dev version of the doc build:

Local dev docs:
![image](https://github.com/inmanta/inmanta-core/assets/98876549/98863ab1-6a7c-418e-a5b3-571eda260455)

**VS**

Stable [docs](https://docs.inmanta.com/inmanta-service-orchestrator/latest/lsm/validation_types/validation_types.html#validation-types):
![image](https://github.com/inmanta/inmanta-core/assets/98876549/ae76a6e6-057b-459b-a31d-12540e6b980c)

The other error:
```sh
inmanta-core/docs/lsm/validation_types/validation_types.rst:67: WARNING: Lexing literal_block 'typedef <attr> as <type> matching self <cmp> <value> [(or | and) <comparison_2> ...]\ntypedef <attr> as <type> matching <value> <cmp> self [(or | and) <comparison_2> ...]' as "inmanta" resulted in an error at token: '|'. Retrying in relaxed mode.
```

Reply:
The lexing here is not done by the compiler itself but by Inmanta sphinx: https://github.com/inmanta/inmanta-sphinx/blob/master/sphinxcontrib/inmanta/pygments_lexer.py.

Reply:
It looks like the syntax highlighter chokes on `'` and `|`.  I found this (discussion)[https://github.com/sphinx-doc/sphinx/issues/4225] about it.
It looks like the prod build just ignores syntax highlighting in case of a lexing error, so even if it pains me to have one block without syntax highlighting here I think this is fine ?

Reply:
This needs to be fixed, but we don't have to do that right now. You can also create a new ticket if it would take too long to fix.

Reply:
I'll open a PR on the lexer to fix this, I didn't know about it until now!